Mage.Sets/src/mage/cards/s/SphinxOfTheRevelation.java

```diff
@@ -0,0 +1,90 @@
+package mage.cards.s;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.constants.SubType;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.abilities.Ability;
+import mage.abilities.common.GainLifeControllerTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.PayVariableEnergyCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.dynamicvalue.common.GetXValue;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.counter.GetEnergyCountersControllerEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.LifelinkAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+
+/**
+ *
+ * @author grimreap124
+ */
+public final class SphinxOfTheRevelation extends CardImpl {
+
+    public SphinxOfTheRevelation(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.ARTIFACT, CardType.CREATURE }, "{3}{W}{U}");
+
+        this.subtype.add(SubType.SPHINX);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(5);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Lifelink
+        this.addAbility(LifelinkAbility.getInstance());
+
+        // Whenever you gain life, you get that many {E}.
+        this.addAbility(new GainLifeControllerTriggeredAbility(new SphinxOfTheRevelationEffect()));
```

Comment:
Created a new DynamicValue to pass the amount of gained life in this trigger to its resulting effects.

`new GetEnergyCountersControllerEffect(SavedGainedLifeValue.MANY)`
should replace the custom effect.

Reply:
Updated